### ISSUE TITLE
Log when ics-openvpn is told by OS to stop a VPN

### ISSUE THREAD

**Issue by AlmogBaku** [schwabe/ics-openvpn#415]
Hey,
I have a new issue which suddenly started: A few seconds after connection established, the VPN got disconnected.

**Device:**
OnePlus one, device model A0001
Android version 5.1.1
Cyanogen OS 12.1-YOG4PAS2QL

**Version:**
OpenVPN for Android 0.6.43 (from play store)

**Log:**

```
2015-11-01 14:28:01 official build 0.6.43 running on oneplus A0001 (MSM8974), Android 5.1.1 (LMY48B) API 22, ABI armeabi-v7a, (oneplus/bacon/A0001:5.1.1/LMY48B/YOG4PAS2QL:user/release-keys)
2015-11-01 14:28:01 Building configuration…
2015-11-01 14:28:03 started Socket Thread
2015-11-01 14:28:04 Current Parameter Settings:
2015-11-01 14:28:04   config = '/data/data/de.blinkt.openvpn/cache/android.conf'
2015-11-01 14:28:04   mode = 0
2015-11-01 14:28:04   show_ciphers = DISABLED
2015-11-01 14:28:04   show_digests = DISABLED
2015-11-01 14:28:04   show_engines = DISABLED
2015-11-01 14:28:04   genkey = DISABLED
2015-11-01 14:28:04   key_pass_file = '[UNDEF]'
2015-11-01 14:28:04   show_tls_ciphers = DISABLED
2015-11-01 14:28:04   connect_retry_max = 5
2015-11-01 14:28:04 Connection profiles [0]:
2015-11-01 14:28:04   proto = udp
2015-11-01 14:28:04   local = '[UNDEF]'
2015-11-01 14:28:04   local_port = '[UNDEF]'
2015-11-01 14:28:04   remote = 'node2.rimoto.net'
2015-11-01 14:28:04   remote_port = '2194'
2015-11-01 14:28:04   remote_float = DISABLED
2015-11-01 14:28:04   bind_defined = DISABLED
2015-11-01 14:28:04   bind_local = DISABLED
2015-11-01 14:28:04   bind_ipv6_only = DISABLED
2015-11-01 14:28:04   connect_retry_seconds = 5
2015-11-01 14:28:04   connect_timeout = 120
2015-11-01 14:28:04   socks_proxy_server = '[UNDEF]'
2015-11-01 14:28:04   socks_proxy_port = '[UNDEF]'
2015-11-01 14:28:04   socks_proxy_retry = DISABLED
2015-11-01 14:28:04   tun_mtu = 1500
2015-11-01 14:28:04   tun_mtu_defined = ENABLED
2015-11-01 14:28:04   link_mtu = 1500
2015-11-01 14:28:04   link_mtu_defined = DISABLED
2015-11-01 14:28:04   tun_mtu_extra = 0
2015-11-01 14:28:04   tun_mtu_extra_defined = DISABLED
2015-11-01 14:28:04   mtu_discover_type = -1
2015-11-01 14:28:04   fragment = 0
2015-11-01 14:28:04   mssfix = 1450
2015-11-01 14:28:04   explicit_exit_notification = 0
2015-11-01 14:28:04 Connection profiles END
2015-11-01 14:28:04   remote_random = DISABLED
2015-11-01 14:28:04   ipchange = '[UNDEF]'
2015-11-01 14:28:04   dev = 'tun'
2015-11-01 14:28:04   dev_type = '[UNDEF]'
2015-11-01 14:28:04   dev_node = '[UNDEF]'
2015-11-01 14:28:04   lladdr = '[UNDEF]'
2015-11-01 14:28:04   topology = 1
2015-11-01 14:28:04   tun_ipv6 = DISABLED
2015-11-01 14:28:04   ifconfig_local = '[UNDEF]'
2015-11-01 14:28:04   ifconfig_remote_netmask = '[UNDEF]'
2015-11-01 14:28:04   ifconfig_noexec = DISABLED
2015-11-01 14:28:04   ifconfig_nowarn = ENABLED
2015-11-01 14:28:04   ifconfig_ipv6_local = '[UNDEF]'
2015-11-01 14:28:04   ifconfig_ipv6_netbits = 0
2015-11-01 14:28:04   ifconfig_ipv6_remote = '[UNDEF]'
2015-11-01 14:28:04 Network Status: CONNECTED LTE to MOBILE uwap.orange.co.il
2015-11-01 14:28:04   shaper = 0
2015-11-01 14:28:04   mtu_test = 0
2015-11-01 14:28:04   mlock = DISABLED
2015-11-01 14:28:04   keepalive_ping = 0
2015-11-01 14:28:04   keepalive_timeout = 0
2015-11-01 14:28:04   inactivity_timeout = 0
2015-11-01 14:28:04   ping_send_timeout = 0
2015-11-01 14:28:04   ping_rec_timeout = 0
2015-11-01 14:28:04   ping_rec_timeout_action = 0
2015-11-01 14:28:04   ping_timer_remote = DISABLED
2015-11-01 14:28:04   remap_sigusr1 = 0
2015-11-01 14:28:04   persist_tun = ENABLED
2015-11-01 14:28:04   persist_local_ip = DISABLED
2015-11-01 14:28:04   persist_remote_ip = DISABLED
2015-11-01 14:28:04   persist_key = DISABLED
2015-11-01 14:28:04   passtos = DISABLED
2015-11-01 14:28:04   resolve_retry_seconds = 1000000000
2015-11-01 14:28:04   resolve_in_advance = ENABLED
2015-11-01 14:28:04   username = '[UNDEF]'
2015-11-01 14:28:04   groupname = '[UNDEF]'
2015-11-01 14:28:04   chroot_dir = '[UNDEF]'
2015-11-01 14:28:04   cd_dir = '[UNDEF]'
2015-11-01 14:28:04   writepid = '[UNDEF]'
2015-11-01 14:28:04   up_script = '[UNDEF]'
2015-11-01 14:28:04   down_script = '[UNDEF]'
2015-11-01 14:28:04   down_pre = DISABLED
2015-11-01 14:28:04   up_restart = DISABLED
2015-11-01 14:28:04   up_delay = DISABLED
2015-11-01 14:28:04   daemon = DISABLED
2015-11-01 14:28:04   inetd = 0
2015-11-01 14:28:04   log = DISABLED
2015-11-01 14:28:04   suppress_timestamps = DISABLED
2015-11-01 14:28:04   machine_readable_output = ENABLED
2015-11-01 14:28:04   nice = 0
2015-11-01 14:28:04   verbosity = 4
2015-11-01 14:28:04   mute = 0
2015-11-01 14:28:04   gremlin = 0
2015-11-01 14:28:04   status_file = '[UNDEF]'
2015-11-01 14:28:04   status_file_version = 1
2015-11-01 14:28:04   status_file_update_freq = 60
2015-11-01 14:28:04   occ = ENABLED
2015-11-01 14:28:04   rcvbuf = 0
2015-11-01 14:28:04   sndbuf = 0
2015-11-01 14:28:04   sockflags = 0
2015-11-01 14:28:04   fast_io = DISABLED
2015-11-01 14:28:04   comp.alg = 2
2015-11-01 14:28:04   comp.flags = 1
2015-11-01 14:28:04   route_script = '[UNDEF]'
2015-11-01 14:28:04   route_default_gateway = '[UNDEF]'
2015-11-01 14:28:04   route_default_metric = 0
2015-11-01 14:28:04   route_noexec = DISABLED
2015-11-01 14:28:04   route_delay = 0
2015-11-01 14:28:04   route_delay_window = 30
2015-11-01 14:28:04   route_delay_defined = DISABLED
2015-11-01 14:28:04   route_nopull = DISABLED
2015-11-01 14:28:04   route_gateway_via_dhcp = DISABLED
2015-11-01 14:28:04   allow_pull_fqdn = DISABLED
2015-11-01 14:28:04   management_addr = '/data/data/de.blinkt.openvpn/cache/mgmtsocket'
2015-11-01 14:28:04   management_port = 'unix'
2015-11-01 14:28:04   management_user_pass = '[UNDEF]'
2015-11-01 14:28:04   management_log_history_cache = 250
2015-11-01 14:28:04   management_echo_buffer_size = 100
2015-11-01 14:28:04   management_write_peer_info_file = '[UNDEF]'
2015-11-01 14:28:04   management_client_user = '[UNDEF]'
2015-11-01 14:28:04   management_client_group = '[UNDEF]'
2015-11-01 14:28:04   management_flags = 4390
2015-11-01 14:28:04   shared_secret_file = '[UNDEF]'
2015-11-01 14:28:04   key_direction = 2
2015-11-01 14:28:04   ciphername_defined = ENABLED
2015-11-01 14:28:04   ciphername = 'BF-CBC'
2015-11-01 14:28:04   authname_defined = ENABLED
2015-11-01 14:28:04   authname = 'SHA1'
2015-11-01 14:28:04   prng_hash = 'SHA1'
2015-11-01 14:28:04   prng_nonce_secret_len = 16
2015-11-01 14:28:04   keysize = 0
2015-11-01 14:28:04   engine = DISABLED
2015-11-01 14:28:04   replay = ENABLED
2015-11-01 14:28:04   mute_replay_warnings = DISABLED
2015-11-01 14:28:04   replay_window = 64
2015-11-01 14:28:04   replay_time = 15
2015-11-01 14:28:04   packet_id_file = '[UNDEF]'
2015-11-01 14:28:04   use_iv = ENABLED
2015-11-01 14:28:04   test_crypto = DISABLED
2015-11-01 14:28:04   tls_server = DISABLED
2015-11-01 14:28:04   tls_client = ENABLED
2015-11-01 14:28:04   key_method = 2
2015-11-01 14:28:04   ca_file = '[[INLINE]]'
2015-11-01 14:28:04   ca_path = '[UNDEF]'
2015-11-01 14:28:04   dh_file = '[UNDEF]'
2015-11-01 14:28:04   cert_file = '[[INLINE]]'
2015-11-01 14:28:04   extra_certs_file = '[UNDEF]'
2015-11-01 14:28:04   priv_key_file = '[[INLINE]]'
2015-11-01 14:28:04   pkcs12_file = '[UNDEF]'
2015-11-01 14:28:04   cipher_list = '[UNDEF]'
2015-11-01 14:28:04   tls_verify = '[UNDEF]'
2015-11-01 14:28:04   tls_export_cert = '[UNDEF]'
2015-11-01 14:28:04   verify_x509_type = 0
2015-11-01 14:28:04   verify_x509_name = '[UNDEF]'
2015-11-01 14:28:04   crl_file = '[UNDEF]'
2015-11-01 14:28:04   ns_cert_type = 0
2015-11-01 14:28:04   remote_cert_ku[i] = 0
2015-11-01 14:28:04   remote_cert_ku[i] = 0
2015-11-01 14:28:04   remote_cert_ku[i] = 0
2015-11-01 14:28:04   remote_cert_ku[i] = 0
2015-11-01 14:28:04   remote_cert_ku[i] = 0
2015-11-01 14:28:04   remote_cert_ku[i] = 0
2015-11-01 14:28:04   remote_cert_ku[i] = 0
2015-11-01 14:28:04   remote_cert_ku[i] = 0
2015-11-01 14:28:04   remote_cert_ku[i] = 0
2015-11-01 14:28:04   remote_cert_ku[i] = 0
2015-11-01 14:28:04   remote_cert_ku[i] = 0
2015-11-01 14:28:04   remote_cert_ku[i] = 0
2015-11-01 14:28:04   remote_cert_ku[i] = 0
2015-11-01 14:28:04   remote_cert_ku[i] = 0
2015-11-01 14:28:04   remote_cert_ku[i] = 0
2015-11-01 14:28:04   remote_cert_ku[i] = 0
2015-11-01 14:28:04   remote_cert_eku = '[UNDEF]'
2015-11-01 14:28:04   ssl_flags = 0
2015-11-01 14:28:04   tls_timeout = 2
2015-11-01 14:28:04   renegotiate_bytes = 0
2015-11-01 14:28:04   renegotiate_packets = 0
2015-11-01 14:28:04   renegotiate_seconds = 3600
2015-11-01 14:28:04   handshake_window = 60
2015-11-01 14:28:04   transition_window = 3600
2015-11-01 14:28:04   single_session = DISABLED
2015-11-01 14:28:04   push_peer_info = DISABLED
2015-11-01 14:28:04   tls_exit = DISABLED
2015-11-01 14:28:04   tls_auth_file = '[[INLINE]]'
2015-11-01 14:28:04   client = ENABLED
2015-11-01 14:28:04   pull = ENABLED
2015-11-01 14:28:04   auth_user_pass_file = '[UNDEF]'
2015-11-01 14:28:04 OpenVPN 2.4-icsopenvpn [git:icsopenvpn_642-62c7d5f90844609f] android-14-armeabi-v7a [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH] [IPv6] built on Oct 25 2015
2015-11-01 14:28:04 library versions: OpenSSL 1.0.2d 9 Jul 2015, LZO 2.09
2015-11-01 14:28:04 MANAGEMENT: Connected to management server at /data/data/de.blinkt.openvpn/cache/mgmtsocket
2015-11-01 14:28:04 MANAGEMENT: CMD 'hold release'
2015-11-01 14:28:04 MANAGEMENT: CMD 'bytecount 2'
2015-11-01 14:28:04 MANAGEMENT: CMD 'proxy NONE'
2015-11-01 14:28:04 MANAGEMENT: CMD 'state on'
2015-11-01 14:28:05 WARNING: No server certificate verification method has been enabled.  See http://openvpn.net/howto.html#mitm for more info.
2015-11-01 14:28:05 Control Channel Authentication: tls-auth using INLINE static key file
2015-11-01 14:28:05 Outgoing Control Channel Authentication: Using 160 bit message hash 'SHA1' for HMAC authentication
2015-11-01 14:28:05 Incoming Control Channel Authentication: Using 160 bit message hash 'SHA1' for HMAC authentication
2015-11-01 14:28:05 LZO compression initializing
2015-11-01 14:28:05 Control Channel MTU parms [ L:1542 D:1184 EF:66 EB:0 ET:0 EL:3 ]
2015-11-01 14:28:05 Data Channel MTU parms [ L:1542 D:1450 EF:42 EB:393 ET:0 EL:3 ]
2015-11-01 14:28:05 Local Options String (VER=V4): 'V4,dev-type tun,link-mtu 1542,tun-mtu 1500,proto UDPv4,comp-lzo,keydir 1,cipher BF-CBC,auth SHA1,keysize 128,tls-auth,key-method 2,tls-client'
2015-11-01 14:28:05 Expected Remote Options String (VER=V4): 'V4,dev-type tun,link-mtu 1542,tun-mtu 1500,proto UDPv4,comp-lzo,keydir 0,cipher BF-CBC,auth SHA1,keysize 128,tls-auth,key-method 2,tls-server'
2015-11-01 14:28:05 TCP/UDP: Preserving recently used remote address: [AF_INET]52.28.186.83:2194
2015-11-01 14:28:05 Socket Buffers: R=[163840->163840] S=[163840->163840]
2015-11-01 14:28:05 MANAGEMENT: CMD 'needok 'PROTECTFD' ok'
2015-11-01 14:28:05 UDP link local: (not bound)
2015-11-01 14:28:05 UDP link remote: [AF_INET]52.28.186.83:2194
2015-11-01 14:28:05 MANAGEMENT: >STATE:1446380885,WAIT,,,
2015-11-01 14:28:05 MANAGEMENT: >STATE:1446380885,AUTH,,,
2015-11-01 14:28:05 TLS: Initial packet from [AF_INET]52.28.186.83:2194, sid=8d073dc4 75326da7
2015-11-01 14:28:05 VERIFY OK: depth=1, C=IL, ST=Merkaz, O=Rimoto, OU=R&D, CN=Rimoto CORE Certificate Authority, emailAddress=dev@rimotosoftware.com
2015-11-01 14:28:05 VERIFY OK: depth=0, C=IL, ST=Merkaz, O=Rimoto, OU=Users, CN=node2.rimoto.net
2015-11-01 14:28:06 Data Channel Encrypt: Cipher 'BF-CBC' initialized with 128 bit key
2015-11-01 14:28:06 Data Channel Encrypt: Using 160 bit message hash 'SHA1' for HMAC authentication
2015-11-01 14:28:06 Data Channel Decrypt: Cipher 'BF-CBC' initialized with 128 bit key
2015-11-01 14:28:06 Data Channel Decrypt: Using 160 bit message hash 'SHA1' for HMAC authentication
2015-11-01 14:28:06 Control Channel: TLSv1, cipher TLSv1/SSLv3 DHE-RSA-AES256-SHA, 2048 bit RSA
2015-11-01 14:28:06 [node2.rimoto.net] Peer Connection Initiated with [AF_INET]52.28.186.83:2194
2015-11-01 14:28:07 MANAGEMENT: >STATE:1446380887,GET_CONFIG,,,
2015-11-01 14:28:07 SENT CONTROL [node2.rimoto.net]: 'PUSH_REQUEST' (status=1)
2015-11-01 14:28:07 PUSH: Received control message: 'PUSH_REPLY,redirect-gateway def1,dhcp-option DNS 10.7.1.1,route 10.50.0.1,topology net30,ping 10,ping-restart 30,ifconfig 10.50.0.22 10.50.0.21'
2015-11-01 14:28:07 OPTIONS IMPORT: timers and/or timeouts modified
2015-11-01 14:28:07 OPTIONS IMPORT: --ifconfig/up options modified
2015-11-01 14:28:07 OPTIONS IMPORT: route options modified
2015-11-01 14:28:07 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified
2015-11-01 14:28:07 ROUTE_GATEWAY 127.100.103.119/255.0.0.0 IFACE=lo HWADDR=00:00:00:00:00:00
2015-11-01 14:28:07 do_ifconfig, tt->ipv6=0, tt->did_ifconfig_ipv6_setup=0
2015-11-01 14:28:07 MANAGEMENT: >STATE:1446380887,ASSIGN_IP,,10.50.0.22,
2015-11-01 14:28:07 MANAGEMENT: CMD 'needok 'IFCONFIG' ok'
2015-11-01 14:28:07 MANAGEMENT: CMD 'needok 'ROUTE' ok'
2015-11-01 14:28:07 MANAGEMENT: >STATE:1446380887,ADD_ROUTES,,,
2015-11-01 14:28:07 MANAGEMENT: CMD 'needok 'ROUTE' ok'
2015-11-01 14:28:07 MANAGEMENT: CMD 'needok 'DNSSERVER' ok'
2015-11-01 14:28:07 MANAGEMENT: CMD 'needok 'PERSIST_TUN_ACTION' OPEN_BEFORE_CLOSE'
2015-11-01 14:28:07 Opening tun interface:
2015-11-01 14:28:07 Local IPv4: 10.50.0.22/30 IPv6: null MTU: 1500
2015-11-01 14:28:07 DNS Server: 10.7.1.1, Domain: null
2015-11-01 14:28:07 Routes: 0.0.0.0/0, 10.50.0.1/32, 10.50.0.20/30 
2015-11-01 14:28:07 Routes excluded:  
2015-11-01 14:28:07 VpnService routes installed: 0.0.0.0/0 
2015-11-01 14:28:07 Disallowed VPN apps: 
2015-11-01 14:28:07 MANAGEMENT: CMD 'needok 'OPENTUN' ok'
2015-11-01 14:28:07 Initialization Sequence Completed
2015-11-01 14:28:07 MANAGEMENT: >STATE:1446380887,CONNECTED,SUCCESS,10.50.0.22,52.28.186.83
2015-11-01 14:28:08 PID_ERR replay-window backtrack occurred [1] [SSL-0] [0_000000] 0:8 0:7 t=1446380888[0] r=[0,64,15,1,1] sl=[56,8,64,272]
2015-11-01 14:28:41 MANAGEMENT: CMD 'signal SIGINT'
2015-11-01 14:28:41 TCP/UDP: Closing socket
2015-11-01 14:28:41 Sorry, deleting routes on Android is not possible. The VpnService API allows routes to be set on connect only.
2015-11-01 14:28:41 Sorry, deleting routes on Android is not possible. The VpnService API allows routes to be set on connect only.
2015-11-01 14:28:41 Closing TUN/TAP interface
2015-11-01 14:28:42 SIGINT[hard,] received, process exiting
2015-11-01 14:28:42 MANAGEMENT: >STATE:1446380922,EXITING,SIGINT,,

```

**Comment by schwabe** [comment]
That is a very strange one. The log indicates that the app would have closed the VPN. Did you manage to grab adb logcat while this was running?

**Comment by AlmogBaku** [comment]
```
11-03 15:52:33.310 23979-24008/de.blinkt.openvpn W/art: Suspending all threads took: 5.879ms
11-03 15:52:34.428 23979-27149/de.blinkt.openvpn I/OpenVPN: Giving up
11-03 15:52:34.462 23979-27149/de.blinkt.openvpn I/OpenVPN: Exiting
```

**Comment by schwabe** [comment]
Hm that adb logcat seems not to be related to the reason why the VPN has been disconnected.

Is this reproducible?

**Comment by AlmogBaku** [comment]
It seems like for some reason Opera Max(also a VPN app) _suddenly_ started to interrupt and caused the problems (although it was disabled).
Deleting the app solved the problem.

**Comment by AlmogBaku** [comment]
ill try to provide more logs to detect that, although I already removed the problematic app.

**Comment by schwabe** [comment]
The app should now log the if it detects this situation.

**Comment by nikoliv** [comment]
I had the same issue. 

OpenVPN Connect (1.1.16) started disconnecting within a minute of establishing the connection. 

I deployed a profile on another android phone and it worked fine so I confirmed it's not server side.

Going back to the problem device (OnePlus one, device model A0001, Android version 5.1.1, Cyanogen OS 12.1-YOG4PAS1N0) I switched to OpenVPN for Android (0.6.46) and got the same error messages in the log upon dropping connection:
"Sorry, deleting routes on Android is not possible. The VpnService API allows routes to be set on connect only."

I had OperaMax in disabled state, which I had recently removed and re-installed again and somewhere at that time my VPN issues started. I removed the app and VPN issues were gone.
